resources/fetchersManager/data.go

```diff
@@ -23,49 +23,40 @@ import (
 	"sync"
 	"time"
 
-	"github.com/elastic/beats/v7/libbeat/common/atomic"
 	"github.com/elastic/cloudbeat/resources/fetching"
 	"github.com/elastic/elastic-agent-libs/logp"
 )
 
 // Data maintains a cache that is updated by Fetcher implementations registered
 // against it. It sends the cache to an output channel at the defined interval.
 type Data struct {
-	log        *logp.Logger
-	timeout    time.Duration
-	interval   time.Duration
-	fetchers   FetchersRegistry
-	stop       chan struct{}
-	stopNotice chan time.Duration
-}
+	log *logp.Logger
+
+	// Maximum duration of a single fetcher
+	timeout time.Duration
 
-type Stop func(context.Context, time.Duration)
+	// Time between two consecutive cycles
+	interval time.Duration
+
+	fetchers FetchersRegistry
+}
 
-// NewData returns a new Data instance.
-// interval is the duration the manager wait between two consecutive cycles.
-// timeout is the maximum duration the manager wait for a single fetcher to return results.
 func NewData(log *logp.Logger, interval time.Duration, timeout time.Duration, fetchers FetchersRegistry) (*Data, error) {
 	return &Data{
-		log:        log,
-		timeout:    timeout,
-		interval:   interval,
-		fetchers:   fetchers,
-		stop:       make(chan struct{}),
-		stopNotice: make(chan time.Duration),
+		log:      log,
+		timeout:  timeout,
+		interval: interval,
+		fetchers: fetchers,
 	}, nil
 }
 
 // Run starts all configured fetchers to collect resources.
-func (d *Data) Run(ctx context.Context) Stop {
+func (d *Data) Run(ctx context.Context) {
 	go d.fetchAndSleep(ctx)
-	called := atomic.NewBool(false)
-	return func(ctx context.Context, grace time.Duration) {
-		defer called.Store(true)
-		if called.Load() {
-			return
-		}
-		d.stopData(ctx, grace)
-	}
+}
+
+func (d *Data) Stop() {
+	d.fetchers.Stop()
```

Comment:
It seems like calling the `Stop` by itself, won't cause the `fetchAndSleep` to return (and it should).